src/main/java/decodes/cwms/CwmsConnectionPool.java

```diff
@@ -0,0 +1,399 @@
+package decodes.cwms;
+
+import java.io.PrintWriter;
+import java.lang.management.ManagementFactory;
+import java.sql.CallableStatement;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.management.JMException;
+import javax.management.ObjectName;
+import javax.management.openmbean.OpenDataException;
+import javax.management.openmbean.TabularData;
+import javax.management.openmbean.TabularDataSupport;
+
+import decodes.db.Constants;
+import decodes.sql.DbKey;
+import decodes.tsdb.BadConnectException;
+import decodes.tsdb.DbIoException;
+import ilex.util.StringPair;
+import ilex.util.TextUtil;
+import opendcs.org.opendcs.jmx.ConnectionPoolMXBean;
+import opendcs.org.opendcs.jmx.connections.JMXTypes;
+import opendcs.util.sql.WrappedConnection;
+import usace.cwms.db.dao.ifc.sec.CwmsDbSec;
+import usace.cwms.db.dao.util.connection.ConnectionLoginInfo;
+import usace.cwms.db.dao.util.connection.CwmsDbConnectionPool;
+import usace.cwms.db.dao.util.services.CwmsDbServiceLookup;
+
+public class CwmsConnectionPool implements ConnectionPoolMXBean
+{
+    private static Logger log = Logger.getLogger(CwmsConnectionPool.class.getName());
+
+    private static HashMap<CwmsConnectionInfo,CwmsConnectionPool> pools = new HashMap<>();
+    private HashSet<WrappedConnection> connectionsOut = new HashSet<>();
+    private CwmsConnectionInfo info = null;	
+	private int connectionsRequested = 0;
+	private int connectionsFreed = 0;
+	private int unknownConnReturned = 0;
+    private static CwmsDbConnectionPool pool = CwmsDbConnectionPool.getInstance();
+
+
+    public static CwmsConnectionPool getPoolFor(CwmsConnectionInfo info) throws BadConnectException
+    {
+        CwmsConnectionPool ret = pools.get(info);
+        if (ret == null)
+        {
+            try(Connection conn = pool.getConnection(info.getLoginInfo());)
+			{
+                ret = new CwmsConnectionPool(info,conn);
+                pools.put(info,ret);        
+            }
+            catch(SQLException ex)
+            {
+                throw new BadConnectException("Unable to initialize pool.",ex);
+            }
+            catch(DbIoException ex)
+            {
+                throw new BadConnectException("Unable to set office context.",ex);
+            }
+        }
+        return ret;
+    }
+
+
+    private CwmsConnectionPool(CwmsConnectionInfo info, Connection conn) throws BadConnectException,SQLException,DbIoException
+    {
+        this.info = info;
```

Comment:
This constructor could use some love. I'd recommend removing all logic from here and moving to the factory method instead. Using one parameter to populate the other is adding a lot to the cognitive load here.